### PR TITLE
framework/ble_manager: apply lwnl in BLE callback process

### PR DESCRIPTION
**Pull request by an4967**
- change BLE callback process from direct call to lwnl handling.
- re-order mac address of scanned data.

### CODE REVIEW

File: framework/src/ble_manager/ble_manager_lwnl_listener.c [hunk 1]
```diff
@@ -92,16 +115,17 @@ int lwnl_fetch_ble_event(int fd, void *buf, int buflen)
...
 		hmsg->msg = &g_msg;
 	} else {
 		/* Remove wifi data in socket. */
 		char *t_buf = (char *)malloc(len);
-		nbytes = read(fd, (char *)t_buf, len);
-		BLE_LOG_INFO("read unused data nbytes : %d / len : %d\n", nbytes, len);
-		free(t_buf);
+		if (t_buf) {
+			nbytes = read(fd, (char *)t_buf, len);
+			free(t_buf);
```
Comment (jeongchanKim) [inline framework/src/ble_manager/ble_manager_lwnl_listener.c:127]: After reading the data into `buf`, free it immediately.
Is this correct?
And if it is correct, why do we need to read?

File: os/net/blemgr/bledev_mgr_client.c [hunk 2]
```diff
@@ -16,54 +16,69 @@
...
 	return;
 }
 
 static void bledrv_operation_notification_cb(trble_operation_handle *handle, trble_data *read_result)
 {
-	if (g_client_config.ble_client_operation_notification_cb != NULL) {
-		g_client_config.ble_client_operation_notification_cb((ble_client_operation_handle *)handle, (ble_data *)read_result);
-	}
+	uint32_t size = sizeof(trble_operation_handle) + sizeof(read_result->length) + read_result->length;
+	uint8_t *data = (uint8_t *)kmm_malloc(size);
```
Comment (jeongchanKim) [inline os/net/blemgr/bledev_mgr_client.c:65]: Is it okay not to check that allocation is success or not?

File: os/net/blemgr/bledev_mgr_client.c [hunk 3]
```diff
@@ -16,54 +16,69 @@
...
 static void bledrv_device_scanned_cb(trble_scanned_device *scanned_device)
 {
-	if (g_client_config.ble_client_device_scanned_cb != NULL) {
-		g_client_config.ble_client_device_scanned_cb((ble_client_scanned_device *)scanned_device);
+	int i;
+	int j;
+	uint8_t temp;
+	uint8_t *mac = scanned_device->addr.bd_addr; // This mac should be re-ordered.
+
+	for (i = 0, j = TRBLE_BD_ADDR_MAX_LEN - 1;i < j; i++, j--) {
```
Comment (jeongchanKim) [inline os/net/blemgr/bledev_mgr_client.c:41]: Please add space after `;`.

File: framework/src/ble_manager/ble_manager_lwnl_listener.c [hunk 4]
```diff
@@ -92,16 +115,17 @@ int lwnl_fetch_ble_event(int fd, void *buf, int buflen)
 	memcpy(&status, type_buf, sizeof(lwnl_cb_status));
 	memcpy(&len, type_buf + sizeof(lwnl_cb_status), sizeof(uint32_t));
 
-	BLE_LOG_VERBOSE("dev state(%d) length(%d)\n", status.evt, len);
+	BLE_LOG_INFO("dev state(%d) length(%d)\n", status.evt, len);
 	if (status.type == LWNL_DEV_BLE) {
 		_lwnl_call_event(fd, status, len);
 		hmsg->msg = &g_msg;
 	} else {
 		/* Remove wifi data in socket. */
```
Comment (jeongchanKim) [inline framework/src/ble_manager/ble_manager_lwnl_listener.c:123]: Is this correct comment? Not ble data in socket?

Comment (an4967) [reply framework/src/ble_manager/ble_manager_lwnl_listener.c:123]: Yes, lwnl descriptor is used by both of WiFi & BLE.
If the data is not for BLE, it should be flushed

File: os/net/blemgr/bledev_mgr_client.c (hunk 2 above)
Comment (an4967) [reply os/net/blemgr/bledev_mgr_client.c:65]: Ok, I will check data is null or not

File: os/net/blemgr/bledev_mgr_client.c (hunk 3 above)
Comment (an4967) [reply os/net/blemgr/bledev_mgr_client.c:41]: I fixed it

**Review by jeongchanKim — DISMISSED** [review]
LGTM

**Review by pillip8282 — APPROVED** [review]
lgtm

**Pull request merged**